src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PotentialEditorConfigDataSource.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    [Export(typeof(IFileWatchProviderDataSource))]
```

Comment:
Something is wrong either with how I'm exporting this type or importing it in the CPS change. The service that consumes this type imports it like so:

``` C#
[ImportMany(ExportContractNames.Scopes.ConfiguredProject, typeof(IFileWatchProviderDataSource))]
private OrderPrecedenceImportCollection<IProjectValueDataSource<IFileWatchProvider>> FileWatchProviderDataSources { get; set; }
```

but it always ends up with an empty collection and I'm not sure why.

Reply:
You are missing an AppliesTo, this should be marked with DotNetLanguageService must likely.

Reply:
I assume you mean I need to add an `AppliesTo` attribute to the `PotentialEditorConfigDataSource` type. I did so, but it is still never being instantiated. Where do I go from here to diagnose that issue?

Reply:
On the CPS side I tried changing 
``` C#
[ImportMany(ExportContractNames.Scopes.ConfiguredProject, typeof(IFileWatchProviderDataSource))]
```
to just
``` C#
[ImportMany(typeof(IFileWatchProviderDataSource))]
```

and that worked. I don't understand what that argument to `ImportMany` does, however.

Reply:
Ah, I missed the import definition.  You will still need [AppliesTo] regardless.

The first argument is the "contract name", the second argument is the "contract type" both have to be the same on both sides for the "contract to match". We pass the scopes (UnconfiguredProject, ConfiguredProject) as a "contract name" when a given contract type (IFileWatchProviderDataSource) might live in [multiple scopes](https://github.com/Microsoft/VSProjectSystem/blob/1c0a47aba5a22d3eb071dc097b73851bdeaf68db/doc/overview/scopes.md).

For example, CPS has a service called `IProjectAsynchronousTasksService` that has two different implementations; one that lives in every configuration, and one that lives in every project. They are distinguished by scope arguments to "contract name".

If the service `IFileWatchProviderDataSource` will live in different scopes, then it should have a scope argument, if not, then it does not need one.